config/config.go

```diff
@@ -171,10 +170,42 @@ func WriteConfigValueToConfigFile(key, value string) error {
 	}
 	configValues[key] = value
 
-	err = WriteYamlFile(f.Name(), &configValues)
+	err = atomicwrite(f.Name(), &configValues)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func atomicwrite(file string, c *ContextMap) (err error) {
+	// create a temp file
+	path, filename := filepath.Split(file)
+
+	f, err := ioutil.TempFile(path, filename) //tempfile name is generated by adding a random string to the end of given filename
+	if err != nil {
+		return fmt.Errorf("cannot create temp file: %v", err)
+	}
+
+	defer f.Close()
+	defer os.Remove(f.Name())
+
+	err = WriteYamlFile(f.Name(), c)
+	if err != nil {
+		return err
+	}
+
+	info, err := os.Stat(file)
+	if err != nil {
+		return err
+	} else {
+		_ = os.Chmod(f.Name(), info.Mode())
```

Comment:
Just a nit, but we ignore the error here. Is it not worth chucking this back too? 

NB :I don't know if it is possible , to set things up in a way that would cause this, but if anyone every want to use the CLI inside a container the sometimes things might not not set up quite right.

Reply:
Just has a look at this: it returns a 'PathError'. It is probably fine.